task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,281 @@
+            fi
+
+            KFP_EXIT_CODE=0
+            ${csfilter_kfp_cmd[@]} processed_sast_unicode_check_out.json > sast_unicode_check_out.json || KFP_EXIT_CODE=$?
+            if [ "${KFP_EXIT_CODE}" -eq 0 ]; then
+                echo "Message: Succeed to filter known false positives" >&2
+            else
+                echo "Error: failed to filter known false positives" >&2
+                return 1
+            fi
+
+            if [ -f "raw-excluded-findings.json" ]; then
+                csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ raw-excluded-findings.json > excluded-findings.json
+            fi
+        }
+
+        generate_sarif_report()
+        {
+            csgrep --mode=sarif sast_unicode_check_out.json > sast_unicode_check_out.sarif
+            if [[ "${FUC_EXIT_CODE}" -eq 0 ]]; then
+                note="Task $(context.task.name) success: No finding was detected"
+                ERROR_OUTPUT=$(make_result_json -r SUCCESS -t "$note")
+            elif [[ "${FUC_EXIT_CODE}" -eq 1 ]] && [[ ! -s  sast_unicode_check_out.sarif ]]; then
+                note="Task $(context.task.name) success: Some findings were detected, but filtered by known false positive"
+                ERROR_OUTPUT=$(make_result_json -r SUCCESS -t "$note")
+            else
+                echo "sast-unicode-check test failed because of the following issues:"
+                cat processed_sast_unicode_check_out.json
+                TEST_OUTPUT=
+                parse_test_output $(context.task.name) sarif sast_unicode_check_out.sarif  || true
+                note="Task $(context.task.name) failed: For details, check Tekton task log."
+                ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            fi
+            echo "${TEST_OUTPUT:-${ERROR_OUTPUT}}" | tee $(results.TEST_OUTPUT.path)
+        }
+
+        main() 
+        {
+            clone_unicontrol
+            unicontrol_scan
+            process_all_results
+            csfilter_kfp
+            generate_sarif_report
+        }
+
+        main
+    - name: upload
+      image: quay.io/konflux-ci/oras:latest@sha256:f4b891ee3038a5f13cd92ff4f473faad5601c2434d1c6b9bccdfc134d9d5f820
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      env:
+        - name: IMAGE_URL
+          value: $(params.image-url)
+        - name: IMAGE_DIGEST
+          value: $(params.image-digest)
+      script: |
+        #!/usr/bin/env bash
+
+        if [ -z "${IMAGE_URL}" ] || [ -z "${IMAGE_DIGEST}"  ]; then
+          echo 'No image-url or image-digest param provided. Skipping upload.'
+          exit 0;
+        fi
+
+        UPLOAD_FILES="sast_unicode_check_out.sarif excluded-findings.json"
+        for UPLOAD_FILE in ${UPLOAD_FILES}; do
+            if [ ! -f "${UPLOAD_FILE}" ]; then
+              echo "No ${UPLOAD_FILE} exists. Skipping upload."
+              continue;
```

Comment:
Please remove the semicolon after `continue`.

Reply:
Removed it